cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java

```diff
@@ -177,6 +177,29 @@ public void testHeapDumpGenerationForOOM() {
         }
     }
 
+    @Test(description = "Check test command is preserving bin jar in target directory")
+    public void testTestCommandPreservingBinJarInTargetDir() {
+        Path projectPath = this.testResources.resolve("validMultiModuleProjectWithTests");
+        System.setProperty(ProjectConstants.USER_DIR, projectPath.toString());
+        // build the project
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+        new CommandLine(buildCommand).parse();
+        buildCommand.execute();
+        Assert.assertTrue(projectPath.resolve("target").resolve("bin").resolve("winery.jar").toFile().exists());
+
+        // Run tests
+        TestCommand testCommand = new TestCommand(projectPath, printStream, printStream, false);
+        new CommandLine(testCommand).parse();
+        testCommand.execute();
+        Assert.assertTrue(projectPath.resolve("target").resolve("bin").resolve("winery.jar").toFile().exists());
```

Comment:
You probably need to check the md5 of the executable jar here before and after test command.

Reply:
Added